workflow/rules/common.smk

```diff
@@ -88,19 +88,29 @@ def get_model(wildcards):
 
 def is_single_end(sample, unit):
     """Determine whether unit is single-end."""
-    bam_paired_not_present = pd.isnull(units.loc[(sample, unit), "bam_paired"])
+    if "bam_paired" in units.columns:
+        bam_paired_not_present = pd.isnull(units.loc[(sample, unit), "bam_paired"])
+    else:
+        bam_paired_not_present = True
     fq2_not_present = pd.isnull(units.loc[(sample, unit), "fq2"])
     return fq2_not_present and bam_paired_not_present
 
 
 def get_fastqs(wildcards):
     """Get raw FASTQ files from unit sheet."""
-    if not pd.isnull(units.loc[(wildcards.sample, wildcards.unit), "bam_single"]):
+    if "bam_single" in units.columns and not pd.isnull(
+        units.loc[(wildcards.sample, wildcards.unit), "bam_single"]
+    ):
         return f"results/fastq/{wildcards.sample}-{wildcards.unit}.fq.gz"
-    elif not pd.isnull(units.loc[(wildcards.sample, wildcards.unit), "bam_paired"]):
-        fqfrombam1 = f"results/fastq/{wildcards.sample}-{wildcards.unit}.1.fq.gz"
-        fqfrombam2 = f"results/fastq/{wildcards.sample}-{wildcards.unit}.2.fq.gz"
-        return [fqfrombam1, fqfrombam2]
+    elif "bam_paired" in units.columns and not pd.isnull(
+        units.loc[(wildcards.sample, wildcards.unit), "bam_paired"]
+    ):
```

Comment:
And this could also use the newly defined function:
```suggestion
    elif column_missing_or_empty("bam_paired", units, wildcards.sample, wildcards.unit):
```